pg_search/src/gucs.rs

```diff
@@ -165,76 +133,89 @@ pub fn per_tuple_cost() -> f64 {
     PER_TUPLE_COST.get()
 }
 
-pub fn create_index_parallelism() -> NonZeroUsize {
-    let nthreads = unsafe {
-        let mut nthreads = pg_sys::max_parallel_maintenance_workers;
-        if pg_sys::parallel_leader_participation {
-            nthreads += 1
-        }
-        nthreads
-    };
-    adjust_nthreads(nthreads)
-}
-
-pub fn create_index_memory_budget() -> usize {
-    let memory_budget = CREATE_INDEX_MEMORY_BUDGET.get();
-    adjust_budget(memory_budget, create_index_parallelism())
-}
-
-// NB:  this is always `1` (one).  There's no (longer) a concept of parallelism during INSERTS/UPDATES
-pub fn statement_parallelism() -> NonZeroUsize {
-    unsafe { NonZeroUsize::new_unchecked(1) }
-}
-
-pub fn statement_memory_budget() -> usize {
-    adjust_budget(STATEMENT_MEMORY_BUDGET.get(), statement_parallelism())
-}
-
-fn adjust_nthreads(nthreads: i32) -> NonZeroUsize {
-    let nthreads = if nthreads <= 0 {
-        std::thread::available_parallelism()
-            .expect("your computer should have at least one core")
-            .get()
-    } else {
-        nthreads as usize
-    };
-
-    unsafe {
-        // SAFETY:  we ensured above that nthreads is > 0
-        NonZeroUsize::new_unchecked(nthreads)
-    }
-}
-
-fn adjust_budget(per_thread_budget: i32, parallelism: Parallelism) -> usize {
+pub fn adjust_mem(mem: usize, nlaunched: usize, min_mem_per_worker: usize) -> usize {
     // NB:  These limits come from [`tantivy::index_writer::MEMORY_BUDGET_NUM_BYTES_MAX`], which is not publicly exposed
     mod limits {
         // Size of the margin for the `memory_arena`. A segment is closed when the remaining memory
         // in the `memory_arena` goes below MARGIN_IN_BYTES.
         pub const MARGIN_IN_BYTES: usize = 1_000_000;
 
-        // We impose the memory per thread to be at least 15 MB, as the baseline consumption is 12MB,
+        // We impose the memory per thread to be at least 64 MB
```

Comment:
Where does this 64MB value come from, and why not our bare minimum of 15MB?

Is the comment just wrong here?